Userland/Libraries/LibCore/CMakeLists.txt

```diff
@@ -6,14 +6,25 @@ set(SOURCES
     ArgsParser.cpp
     Directory.cpp
     DirectoryEntry.cpp
-    DirIterator.cpp
     Environment.cpp
     File.cpp
     StandardPaths.cpp
     System.cpp
     Version.cpp
 )
 
+if (WIN32)
+    list(APPEND SOURCES
+        DirIteratorWindows.cpp
+    )
+else()
+    list(APPEND SOURCES
+        DirIterator.cpp
+    )
+endif()
+
+add_cxx_compile_options(-Wno-unneeded-internal-declaration)
```

Comment:
why is this flag needed? if we don't need the decl we should remove the function.

Reply:
Problems with constexpr arose when I moved functions: [link](https://tio.run/##ZU5BDsIgELzzij22MVy8mr7DY4OwVBK6EFiMpunXRVu10Tinmd2ZyegYpfaKhlpJjZij0giTuARnwDbt4cWGhW3HaRazEJkVOw06UGa8xgSnEDwYl5C4P6vcm55vEaEDTgWfcUcMo3LUtGIS8IazXw3NX7rdjAvWQR@BPuPPdx0513rX1qshV5nZdHq326sqj5hSSF2hktFIW0izC/QA). This is also the reason why g++ CI fails.

Reply:
Are you OK with DirectoryEntry.h being
```cpp
/*
 * Copyright (c) 2023, Sam Atkins <atkinssj@serenityos.org>
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#pragma once

#include <AK/ByteString.h>
#include <AK/StringView.h>
#if defined(AK_OS_WINDOWS)
typedef struct _WIN32_FIND_DATAA WIN32_FIND_DATA;
#else
#    include <dirent.h>
#endif

namespace Core {

struct DirectoryEntry {
    enum class Type {
        BlockDevice,
        CharacterDevice,
        Directory,
        File,
        NamedPipe,
        Socket,
        SymbolicLink,
        Unknown,
        Whiteout,
    };
    Type type;
    // FIXME: Once we have a special Path string class, use that.
    ByteString name;
#if !defined(AK_OS_WINDOWS)
    ino_t inode_number;

    static StringView posix_name_from_directory_entry_type(Type);
    static StringView representative_name_from_directory_entry_type(Type);
    static Type directory_entry_type_from_stat(mode_t st_mode);
    static DirectoryEntry from_dirent(dirent const&);
    static DirectoryEntry from_stat(DIR*, dirent const&);
#else
    static DirectoryEntry from_find_data(WIN32_FIND_DATA const&);
#endif
};

}
```
Or should I move platform-specific functions from DirectoryEntry to DirIterator to minimize #ifdefs in DirectoryEntry.h?
I don't want to do this because I don't want to deal with the above error.

Reply:
I suppose the version in your comment is fine for now. Though I would prefer `using WIN32_FIND_DATA = _WIN32_FIND_DATAA;` . i.e. using decl vs typedef. Is there a reason that type alias needs to live in the global namespace? or could it live in Core:: or be a nested type of DirectoryEntry?

Reply:
No, it gives error `member access into incomplete type`